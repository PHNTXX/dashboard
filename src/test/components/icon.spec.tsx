import { fireEvent, render, screen } from "@testing-library/react";
import Icon, { IconButton } from "../../components/icon";

const props = {
  name: "bug_report",
  size: "20px",
  onClick: jest.fn(),
};

it("Icon test", () => {
  const { asFragment } = render(<Icon name={props.name} size={props.size} />);
  expect(asFragment).toMatchSnapshot();
});

it("Icon test (no size)", () => {
  const { asFragment } = render(<Icon name={props.name} />);
  expect(asFragment).toMatchSnapshot();
});

it("IconButton test", () => {
  const iconButton = render(
    <IconButton icon={props.name} onClick={props.onClick} />,
  );

  expect(iconButton.asFragment).toMatchSnapshot();

  fireEvent.click(screen.getByText(/bug_report/i));
  expect(props.onClick).toHaveBeenCalledTimes(1);
});
