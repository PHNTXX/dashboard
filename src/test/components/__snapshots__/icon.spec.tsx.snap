// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon test (no size) 1`] = `[Function]`;

exports[`Icon test 1`] = `[Function]`;

exports[`IconButton test 1`] = `[Function]`;
